File: Java/Easy/Accepted/next-greater-element-i_7.java
class Solution {
    public int[] nextGreaterElement(int[] nums1, int[] nums2) {
        
        int[] ans = new int[nums1.length];
        for(int i = 0; i < nums1.length; i++){
            ans[i] = -1;
            int j = 0;
            while(nums2[j] != nums1[i]){
                j++;
            }
            for(int k = j + 1; k < nums2.length; k++){
                if(nums2[k] > nums1[i]){
                    ans[i] = nums2[k];
                    break;
                }
            }
        }
        return ans;
    }
}

File: C++/Easy/Accepted/next-greater-element-i_7.cpp
class Solution {
public:
    vector<int> nextGreaterElement(vector<int>& nums1, vector<int>& nums2) {
        
        unordered_map<int, int> map;
        stack<int> st;
        for(int i = 0; i < nums2.size(); i++) {
            while(!st.empty() && nums2[i] > st.top()) {
                map[st.top()] = nums2[i];
                st.pop();
            }
            st.push(nums2[i]);
        }
        while(!st.empty()) {
            map[st.top()] = -1;
            st.pop();
        }
        vector<int> res;
        for(int i = 0; i < nums1.size(); i++) {
            res.push_back(map[nums1[i]]);
        }
        return res;
    }
};